flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/StreamPlanner.scala

```diff
@@ -102,7 +102,7 @@ class StreamPlanner(
   override def parse(stmt: String): JList[Operation] = {
     val planner = getFlinkPlanner
     // parse the sql query
-    val parsed = planner.parse(stmt)
+    val parsed = planningConfigurationBuilder.createCalciteParser().parse(stmt)
```

Comment:
I would leave a comment here that we don't cache the parser on purpose in oder to configure it per statement, right?